ssl/quic/quic_stream_map.c

```diff
@@ -746,6 +761,43 @@ void ossl_quic_stream_map_gc(QUIC_STREAM_MAP *qsm)
     }
 }
 
+static int eligible_for_shutdown_flush(QUIC_STREAM *qs)
+{
+    /*
+     * We only care about servicing the send part of a stream (if any) during
+     * shutdown flush. A stream needs to have a final size and that final size
+     * needs to be a result of normal conclusion of a stream via
+     * SSL_stream_conclude (as opposed to due to reset). If the application did
+     * not conclude a stream before deleting it we assume it does not care about
+     * data being flushed during connection termination.
```

Comment:
I wonder if an alternative approach is to cause SSL_shutdown() to automatically call SSL_stream_conclude() on any still open streams. This is perhaps more backwards compatible with the way a TLS application works - since there is no SSL_stream_conclude() there.

Reply:
Ok - that seems like a reasonable middle ground. In that case, @t8m's original comment about updating documentation is definitely relevant.

Reply:
I initially thought it would be a good idea to let applications abandon streams they don't want to finish. But on second thought I think I agree here. I'll make this change.

If an application wants to abandon a stream it can just reset it, and/or we can have an API to let them set a no-flush flag on a specific stream if they want to do it without sending a stream reset &mdash; if it turns out anyone wants this.